deps/dicer/lib/Dicer.js

```diff
@@ -175,9 +175,10 @@ Dicer.prototype._oninfo = function(isMatch, data, start, end) {
     this._part._read = function(n) {
       self._unpause();
     };
-    ev = this._isPreamble ? 'preamble' : 'part';
-    if (this._events[ev])
-      this.emit(ev, this._part);
+    if (this._isPreamble && this._events['preamble'])
```

Comment:
is it faster or slower, if there is one top-level if `if (this._isPreamble` and then another if about this._event inside of it?

Reply:
isPreamble is always set to true or false. this._event[ev] could not exist.